server/controllers/csv/csv-controller.js

```diff
@@ -1,8 +1,8 @@
-const Customer = require('../../models/Customer')
+const models = require('../../models')
 const handleCSV = require('../../utilities/handle-csv')
 
 // Upload and save CSV file to database
-const upload = async (req, res) => {
+const uploadCSV = async (req, res) => {
```

Comment:
I should have caught this earlier. I see you are passing to the controller and then the utils function `handle-csv` the arguments request and response. The **request**, and the **response** values should be handled by the routes. The goal of this controller and the utils function is to just indicate if the file was uploaded successfully and the customers saved in the DB.
Remember the single principle, the controller is not responsible on knowing what the payload from the request is, that is the responsibility of the route, where it knows what is in the payload, and is able to extract those values to send them to the controller.
Based on the response the route gets, we want to then return the correct **response**.
So how would you change this?
Instead of returning a `res.status(500).send("Could not upload the file: " + req.file.originalname)`, you can throw a new exception with a new message, then the router can handle the exception returning the correct status, with the message from the exception.
Same with if it succeeds you will return a promise that resolves, and the router will proceed to return a 200 status.

Reply:
Would you be able to help me understand how to refactor this? My understanding is that the controller's purpose is to get the requested data and then return to the browser the needed response. The route's purpose is to take in the request and forward it to the correct controller function. So in this case I can understand that I should probably not be passing in the req, res values into the utility function - but they should still be handled by the controller. Unless I am misunderstanding

Reply:
I made a new commit where I attempted to refactor my approach. The utility function does not return a response now, just a promise. The controller function takes that promise and send a response accordingly.

Reply:
Yes, sorry for not clarifying this:
So an example of the refactoring would be the following, I am going to use the route `getCustomers` as an example:
**Your implementation**
**routes/customer-routes.js:**
```
router.get('/', getCustomers);
```
**customers/customer-controller.js**
```
const getCustomers = (req, res) => {
  models.Customer.findAll()
    .then(data => res.send(data))
    .catch(err => {
      res.status(500).send(err.message || "Some error occured while retrieving customers")
    })
}
```

**With the refactoring:**
The goal of the routes is to destructure and obtain the parameters from the request, and return in the request all the customers.
**routes/customer-routes.js:**
```
router.get('/', (req, res) => {
   getCustomers()
      .then((customers) => {
          res.send(customers)
      })
      .catch((e) => {
          res.status(400).send(e.message)
      })
});
```
Do you see the difference? The controller is not responsible for handling the response of the request, instead it is just responsible from ensuring to give to the route the correct values from the db using the models.
**customers/customer-controller.js**
```
const getCustomers = () => {
  models.Customer.findAll()
}
```

This is an example just with the `resonse`, but the same should be with the request parameters. The routes is responsible for stripping the parameters from the request body, and pass the expected values to the controller.